src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs

```diff
@@ -118,6 +118,12 @@ public class PipelineArtifactServer
                         throw new InvalidOperationException($"Invalid {nameof(downloadParameters.ProjectRetrievalOptions)}!");
                     }
 
+                    IEnumerable<BuildArtifact> fileShareArtifacts = artifacts.Where(a => a.Resource.Type.ToLower() == PipelineArtifactConstants.FileShareArtifact);
+                    if(fileShareArtifacts.Any()) 
+                    {
+                        throw new InvalidOperationException("File Share is not supported in the Download Pipeline Artifact V1 task. Please use Download Pipeline Artifact V2 task instead.");
+                    }
```

Comment:
how is this supposed to only throw on V1 and not V2, this is being used in both scenarios